plugin-server/src/main/pluginsServer.ts

```diff
@@ -423,29 +447,21 @@ export async function startPluginsServer(
             httpServer = createHttpServer(serverConfig.HTTP_SERVER_PORT, healthChecks, analyticsEventsIngestionConsumer)
         }
 
-        // If session recordings consumer is defined, then join it. If join
-        // resolves, then the consumer has stopped and we should shut down
-        // everything else. Ideally we would also join all the other background
-        // tasks as well to ensure we stop the server if we hit any errors and
-        // don't end up with zombie instances, but I'll leave that refactoring
-        // for another time. Note that we have the liveness health checks
-        // already, so in K8s cases zombies should be reaped anyway, albeit not
-        // in the most efficient way.
-        //
-        // When extending to other consumers, we would want to do something like
-        //
-        // ```
-        // try {
-        //      await Promise.race([sessionConsumer.join(), analyticsConsumer.join(), ...])
-        // } finally {
-        //      await closeJobs()
-        // }
-        // ```
-        if (joinSessionRecordingEventsConsumer) {
-            joinSessionRecordingEventsConsumer().catch(closeJobs)
-        }
-        if (joinSessionRecordingBlobConsumer) {
-            joinSessionRecordingBlobConsumer().catch(closeJobs)
+        // If the session recordings or analytics consumer is defined, then join them. If join
+        // resolves, then the consumer has stopped and we should shut down everything else. Ideally
+        // we would also join all the other background tasks as well to ensure we stop the server if
+        // we hit any errors and don't end up with zombie instances, but I'll leave that refactoring
+        // for another time. Note that we have the liveness health checks already, so in K8s cases
+        // zombies should be reaped anyway, albeit not in the most efficient way.
+        const joinPromiseFns = [
+            joinSessionRecordingBlobConsumer,
+            joinAnalyticsEventsIngestionConsumer,
+            joinAnalyticsEventsIngestionHistoricalConsumer,
+            joinAnalyticsEventsIngestionOverflowConsumer,
+        ].filter(Boolean) // Skip any that are undefined.
+
+        if (joinPromiseFns.length > 0) {
+            Promise.race(joinPromiseFns.map((p) => p?.())).catch(closeJobs)
```

Comment:
I wasted a lot of time `await`ing this -- looks like we have pre-existing assumptions here. So just "detaching" the promise here (exactly like it was done before) seems to be the best option.

Reply:
Indeed, we don't want to `await` these, but chain the `.catch(closeJobs)` and leave it unhandled.
Do we need to collect these in `joinPromiseFns` and these local variables, or could we introduce a `shutdownOnConsumerErrors` helper, that we call in the `if capacity` branches? It would handle the logging + call to closeJobs

Reply:
Indeed, we don't want to `await` these, but chain the `.catch(closeJobs)` and leave it unhandled.
Do we need to collect these in `joinPromiseFns` and these local variables, or could we introduce a `shutdownOnConsumerErrors` helper, that we call in the `if capacity` branches? It would handle the logging + call to closeJobs